src/fsharp/symbols/Exprs.fs

```diff
@@ -197,6 +197,62 @@ module FSharpExprConvert =
         | Expr.Op(TOp.ValFieldSet rfref, _, _, _)  when IsStaticInitializationField rfref -> Some ()
         | _ -> None
 
+    let (|ILUnaryOp|_|) e = 
+        match e with 
+        | AI_neg -> Some mkCallUnaryNegOperator
+        | AI_not -> Some mkCallUnaryNotOperator
+        | _ -> None
+
+    let (|ILBinaryOp|_|) e = 
+        match e with 
+        | AI_add
+        | AI_add_ovf
+        | AI_add_ovf_un -> Some mkCallAdditionOperator
+        | AI_sub
+        | AI_sub_ovf
+        | AI_sub_ovf_un -> Some mkCallSubtractionOperator
+        | AI_mul
+        | AI_mul_ovf
+        | AI_mul_ovf_un -> Some mkCallMultiplyOperator
+        | AI_div
+        | AI_div_un     -> Some mkCallDivisionOperator
+        | AI_rem
+        | AI_rem_un     -> Some mkCallModulusOperator
+        | AI_ceq        -> Some mkCallEqualsOperator
+        | AI_clt
+        | AI_clt_un     -> Some mkCallLessThanOperator
+        | AI_cgt
+        | AI_cgt_un     -> Some mkCallGreaterThanOperator
+        | AI_and        -> Some mkCallBitwiseAndOperator
+        | AI_or         -> Some mkCallBitwiseOrOperator
+        | AI_xor        -> Some mkCallBitwiseXorOperator
+        | AI_shl        -> Some mkCallShiftLeftOperator
+        | AI_shr
+        | AI_shr_un     -> Some mkCallShiftRightOperator
+        | _ -> None
+
+    let (|ILConvertOp|_|) e = 
+        match e with 
+        | AI_conv basicTy
+        | AI_conv_ovf basicTy
+        | AI_conv_ovf_un basicTy ->
+            match basicTy with
+            | DT_R  -> Some mkCallToDoubleOperator
+            | DT_I1 -> Some mkCallToSByteOperator
+            | DT_U1 -> Some mkCallToByteOperator
+            | DT_I2 -> Some mkCallToInt16Operator
+            | DT_U2 -> Some mkCallToUInt16Operator
```

Comment:
I'm not convinced expressions such as those generated by 

                 when ^T : int16       and ^U : int16      = (# "conv.ovf.i2" (# "add.ovf" x y : int32 #) : int16 #)

are going to work out right here.  The add.ovf will become a checked int16+int16->int16 addition, then the conv.ovf.i2 will become an  **int16** to **int16** conversion, which is a nop. I suppose that is oddly correct - the checked addition would be enough - but it feels better to code the inversion so that these instructions are treated as a logical pair,

In any case, systematic testing would be appreciated for these.

Reply:
@dsyme added checked operators